runtime/src/iree/hal/drivers/vulkan/vulkan_headers.h

```diff
@@ -41,10 +41,10 @@
 #define VK_USE_PLATFORM_WIN32_KHR 1
 #endif  // IREE_PLATFORM_*
 
-#include "third_party/vulkan_headers/include/vulkan/vulkan.h"  // IWYU pragma: export
+#include "vulkan/vulkan.h"  // IWYU pragma: export
```

Comment:
We want the path to be explicit, so system headers are not used. Maybe we can drop https://github.com/openxla/iree/blob/699b34ca9ca325e47cd709ec9678e8f9b5c89c4a/build_tools/third_party/vulkan_headers/BUILD.overlay#L19 instead? I did add that in https://github.com/openxla/iree/pull/13346, so it's probably required.

Reply:
Interesting. But somehow, my local external bazel build with the IREE runtime only failed to find the header.

Reply:
Can you share more details about your setup and include logs?

Reply:
IIUC, the new local repo for vulkan headers has the root at `third_party/vulkan_headers` and BUILD.overlay assumes the root path. The purpose of `include_prefix` is really to shorten the path from the root path, not to include the parent directories of the root path.

If we really want to include the parent path, we might need to set it up the root path as the iree path and selectively include the files with a new glob pattern, `third_party/vulkan_headers/include/vk_video/*.h`.

Here is the log.
```bash
 /usr/bin/gcc -U_FORTIFY_SOURCE -fstack-protector -Wall -Wunused-but-set-parameter -Wno-free-nonheap-object -fno-omit-frame-pointer -g -MD -MF bazel-out/k8-dbg/bin/external/iree_core/runtime/src/iree/hal/drivers/_objs/drivers/init.pic.d '-frandom-seed=bazel-out/k8-dbg/bin/external/iree_core/runtime/src/iree/hal/drivers/_objs/drivers/init.pic.o' -fPIC '-DIREE_HAVE_HAL_EXECUTABLE_LOADER_EMBEDDED_ELF=1' '-DIREE_HAVE_HAL_EXECUTABLE_LOADER_SYSTEM_LIBRARY=1' -DIREE_HAVE_VMVX_MODULE '-DIREE_HAVE_HAL_EXECUTABLE_LOADER_VMVX_MODULE=1' '-DIREE_HAVE_HAL_EXECUTABLE_EMBEDDED_ELF_PLUGIN=1' '-DIREE_HAVE_HAL_EXECUTABLE_SYSTEM_LIBRARY_PLUGIN=1' '-DIREE_HAVE_HAL_LOCAL_SYNC_DRIVER_MODULE=1' '-DIREE_HAVE_HAL_LOCAL_TASK_DRIVER_MODULE=1' '-DIREE_HAVE_HAL_VULKAN_DRIVER_MODULE=1' '-DBAZEL_CURRENT_REPOSITORY="iree_core"' -iquote external/iree_core -iquote bazel-out/k8-dbg/bin/external/iree_core -iquote external/com_github_dvidelabs_flatcc -iquote bazel-out/k8-dbg/bin/external/com_github_dvidelabs_flatcc -iquote external/cpuinfo -iquote bazel-out/k8-dbg/bin/external/cpuinfo -iquote external/vulkan_headers -iquote bazel-out/k8-dbg/bin/external/vulkan_headers -Ibazel-out/k8-dbg/bin/external/com_github_dvidelabs_flatcc/_virtual_includes/parsing -Ibazel-out/k8-dbg/bin/external/cpuinfo/_virtual_includes/cpuinfo -isystem external/iree_core/runtime/src -isystem bazel-out/k8-dbg/bin/external/iree_core/runtime/src -isystem external/com_github_dvidelabs_flatcc/include -isystem bazel-out/k8-dbg/bin/external/com_github_dvidelabs_flatcc/include -isystem external/cpuinfo/include -isystem bazel-out/k8-dbg/bin/external/cpuinfo/include -isystem external/cpuinfo/src -isystem bazel-out/k8-dbg/bin/external/cpuinfo/src -isystem external/vulkan_headers/include -isystem bazel-out/k8-dbg/bin/external/vulkan_headers/include -Wno-all -Wno-extra -Wno-deprecated -Wno-deprecated-declarations -Wno-ignored-attributes -Wno-array-bounds -Wunused-result '-Werror=unused-result' -Wswitch '-Werror=switch' '-Wno-error=unused-but-set-variable' -DAUTOLOAD_DYNAMIC_KERNELS -fno-canonical-system-headers -Wno-builtin-macro-redefined '-D__DATE__="redacted"' '-D__TIMESTAMP__="redacted"' '-D__TIME__="redacted"' -c external/iree_core/runtime/src/iree/hal/drivers/init.c -o bazel-out/k8-dbg/bin/external/iree_core/runtime/src/iree/hal/drivers/_objs/drivers/init.pic.o)
# Configuration: 727df3c6145fbfabead66e7d67130fb885ff10bc6ad8aabeff18e4017f92861f
# Execution platform: @local_execution_config_platform//:platform
ERROR: /usr/local/google/home/okwan/.cache/bazel/_bazel_okwan/db49338c8767586915e5ab867b37f73b/external/iree_core/runtime/src/iree/hal/drivers/BUILD.bazel:51:24: Compiling runtime/src/iree/hal/drivers/init.c failed: (Exit 1): gcc failed: error executing command (from target @iree_core//runtime/src/iree/hal/drivers:drivers) /usr/bin/gcc -U_FORTIFY_SOURCE -fstack-protector -Wall -Wunused-but-set-parameter -Wno-free-nonheap-object -fno-omit-frame-pointer -g -MD -MF ... (remaining 72 arguments skipped)

Use --sandbox_debug to see verbose messages from the sandbox and retain the sandbox build root for debugging
In file included from external/iree_core/runtime/src/iree/hal/drivers/vulkan/registration/driver_module.h:11,
                 from external/iree_core/runtime/src/iree/hal/drivers/init.c:22:
external/iree_core/runtime/src/iree/hal/drivers/vulkan/vulkan_headers.h:44:10: fatal error: third_party/vulkan_headers/include/vulkan/vulkan.h: No such file or directory
   44 | #include "third_party/vulkan_headers/include/vulkan/vulkan.h"  // IWYU pragma: export
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
```

Reply:
Ah... so the IREE standalone Bazel build works, but IREE as a Bazel dependency in another project does not? That's tricky... wish we had test coverage for it somehow.

and that broke with this recent diff:

```diff
 cc_library(
     name = "vulkan_headers",
-    hdrs = glob(["include/vulkan/*.h"]),
+    hdrs = glob([
+        "include/vk_video/*.h",
+        "include/vulkan/*.h",
+    ]),
     include_prefix = "third_party/vulkan_headers",
+    includes = ["include"],
 )
```

Reply:
The change in BUILD.overlay is fine, but the actual header inclusion prefixed with `third_party/vulkan_headers` is not, and this PR reverts that.

Reply:
The header include path is intentional - see the reasoning here: https://github.com/openxla/iree/blob/60a2566f26b82fb8dd736d5468623b9b7a898a67/build_tools/cmake/iree_copts.cmake#L48-L54

Do you have a branch or something somewhere where someone else can poke at this error? A template repo using Bazel (akin to https://github.com/iree-org/iree-template-runtime-cmake/) could be useful, though I don't really want to advertise support for Bazel like that...

Reply:
The reasoning sounds good and understandable but it is not really compatible with how Bazel works IIUC. It means we always need to expose the iree root as -isystem, but it looks like a hack (especially when we want to make the build component as clean and independently composable as possible.)

Let me check if I can create a smaller repro.